common/src/chain/transaction/signature/mod.rs

```diff
@@ -266,6 +268,7 @@ impl Transactable for SignedTransaction {
 
 fn stream_signature_hash<T: Signable>(
     tx: &T,
+    inputs_utxos: &[&TxOutput],
```

Comment:
wouldn't `&[TxOutput]` be simpler to work with?

Reply:
Yes, but this removes the necessity to copy values, since these outputs aren't generally in one Vec but from multiple transactions.

Reply:
ahh, I see. It's just in all places these outputs are already collected. But ok

Reply:
`TxOutput` is only 24 bytes, I don't think it's worth the troubles to collect references here (and we would need to keep those transactions/UTXO somewhere).

Reply:
The reason why I preferred to do this, besides the size issue, which is minor for this single function, is that this is called through a series of functions and all the other parameters are by reference. I didn't want to break that pattern.